packages/common/lib/types/Authorization.types.ts

```diff
@@ -68,7 +105,7 @@ export interface AuthorizationRequestOpts {
   clientId: string;
   codeChallenge: string;
   codeChallengeMethod: CodeChallengeMethod;
-  authorizationDetails?: AuthorizationDetails[];
+  authorizationDetails?: CommonAuthorizationDetails[];
```

Comment:
I rather see a union here. Problem is that I rather force a developer to accomodate for multiple options, but knowing which properties are available, then having a super interface, and a developer missing certain properties that are present in the subclass/interface

Reply:
added a union type for `AuthorizationDetails`.